### PR TITLE
Niganesh/transform to verifedid datamodels

### PR DESCRIPTION
**Pull request by nithyaganeshng**
Convert the OpenId4VCI network models to Verified ID models.

**Validation:**
Added unit tests.

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandler.kt [hunk 1]
```diff
@@ -71,7 +67,99 @@ internal class OpenId4VCIRequestHandler(private val libraryConfiguration: Librar
...
+        // Get only the supported credential configuration ids from the credential metadata from the list in credential offer.
+        val configIds = credentialOffer.credential_configuration_ids
+        val supportedCredentialConfigurationIds =
+            credentialMetadata.getSupportedCredentialConfigurations(configIds)
+        if (supportedCredentialConfigurationIds.isEmpty())
+            throw OpenId4VciValidationException(
+                "Request does not contain supported credential configuration.",
+                VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value
+            )
+        return supportedCredentialConfigurationIds.first()
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandler.kt:105]: Is there a case when we would need to support more than one?

File: walletlibrary/src/test/java/com/microsoft/walletlibrary/networking/entities/openid4vci/credentialmetadata/LocalizedDisplayDefinitionTest.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        // Act
+        val result = localizedDisplayDefinition.transformToVerifiedIdLogo()
+
+        // Assert
+        assertThat(result.url).isEqualTo("")
+        assertThat(result.altText).isEqualTo("alt_text")
+    }
+
+    @Test
+    fun transformToVerifiedIdLogoNullAltTestForLogo_ReturnsVerifiedIdLogoWithEmptyAltText() {
```
Comment (symorton) [inline walletlibrary/src/test/java/com/microsoft/walletlibrary/networking/entities/openid4vci/credentialmetadata/LocalizedDisplayDefinitionTest.kt]: nit: naming is missing an _

File: walletlibrary/src/test/java/com/microsoft/walletlibrary/networking/entities/openid4vci/credentialmetadata/LocalizedDisplayDefinitionTest.kt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        assertThat(result.backgroundColor).isEqualTo("background_color")
+        assertThat(result.textColor).isEqualTo("text_color")
+        assertThat(result.description).isEqualTo("description")
+        assertThat(result.logo).isNotNull
+        assertThat(result.logo?.url).isEqualTo("uri")
+        assertThat(result.logo?.altText).isEqualTo("alt_text")
+    }
+
+    @Test
+    fun transformToVerifiedIdStyle_nullLogo_ReturnsBasicVerifiedIdStyleWithEmptyLogoValues() {
```
Comment (symorton) [inline walletlibrary/src/test/java/com/microsoft/walletlibrary/networking/entities/openid4vci/credentialmetadata/LocalizedDisplayDefinitionTest.kt]: nit: should nullLogo be capitalized like the tests above. Same question as one below.

File: walletlibrary/src/test/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandlerTest.kt [hunk 4]
```diff
@@ -116,7 +133,259 @@ class OpenId4VCIRequestHandlerTest {
...
+            assertThat(actualException).isInstanceOf(OpenId4VciValidationException::class.java)
+            assertThat(actualException?.message).contains("Request does not contain supported credential configuration.")
+            assertThat((actualException as OpenId4VciValidationException).code).isEqualTo(
+                VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value
+            )
+        }
+    }
+
+    @Test
+    fun handleRequestTest_validVerifiedIdTransformations_ReturnsVerifiedIdRequest() {
```
Comment (symorton) [inline walletlibrary/src/test/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandlerTest.kt]: nit: Valid

**Review by symorton — DISMISSED** [review]

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandler.kt (hunk 1 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/handlers/OpenId4VCIRequestHandler.kt:105]: May be when we support bulk issuance or nested issuance.

**Review by symorton — APPROVED** [review]

**Pull request merged**